Dockerfile

```diff
@@ -1,76 +1,61 @@
-FROM alpine:3.11 AS builder
+FROM docker.io/library/ubuntu:jammy AS builder
 
-RUN apk add --no-cache \
-      bash \
-      bubblewrap \
-      build-base \
-      ca-certificates \
-      coreutils \
-      curl \
-      git \
-      m4 \
-      ncurses-dev \
-      opam \
-      ocaml \
-      ocaml-compiler-libs \
-      patch \
-      rsync \
-      tar \
-      xz \
-    ;
+WORKDIR /tmp
 
-# Disable sandboxing
 RUN set -ex; \
-    echo 'wrap-build-commands: []' > ~/.opamrc; \
-    echo 'wrap-install-commands: []' >> ~/.opamrc; \
-    echo 'wrap-remove-commands: []' >> ~/.opamrc; \
-    echo 'required-tools: []' >> ~/.opamrc;
+    apt-get update; \
+    apt-get install -y --no-install-recommends \
+      wget ca-certificates curl build-essential unzip; \
+    wget https://github.com/coq/platform/archive/refs/tags/2022.04.1.tar.gz; \
+    tar xvf 2022.04.1.tar.gz; \
+    rm -vf 2022.04.1.tar.gz;
+
+WORKDIR /tmp/platform-2022.04.1
 
 ENV OPAMROOT=/opt/coq \
     OPAMYES=1 \
-    OCAML_VERSION=default
+    OCAML_VERSION=default \
+    COQ_PLATFORM_EXTENT=f \
+    COQ_PLATFORM_PACKAGE_PICK_FILE=package_picks/package-pick-8.15~2022.04.sh \
+    COQ_PLATFORM_COMPCERT=n \
+    COQ_PLATFORM_PARALLEL=p \
+    COQ_PLATFORM_JOBS=4 \
+    COQ_PLATFORM_SET_SWITCH=y
 
 RUN set -ex; \
-    mkdir -p /opt/coq; \
-    opam init --no-setup -j 4; \
-    opam repo add coq-released http://coq.inria.fr/opam/released; \
-    opam install -j 4 coq.8.12.0; \
-    opam pin add coq 8.12.0; \
-    opam install \
-      -j 4 \
-      coq-equations.1.2.3+8.12 \
-      coq-mathcomp-ssreflect.1.11.0 \
-    ;
+    echo "" > stdin; \
+    echo "d" >> stdin; \
+    ./coq_platform_make.sh < stdin;
```

Comment:
Let's create a custom `package-pick.sh` based on https://github.com/coq/platform/blob/main/package_picks/package-pick-8.15%7E2022.04.sh by removing packages like `coqide` that's clearly unnecessary for CodeRunner. We can avoid installing many system dependencies that way.
Users can use the original `package-pick` to set up locally.

Also, run `opam clean` to reduce the size of `/opt/coq` after `./coq_platform_make.sh`.

This should reduce the image size significantly.